tests/unit/test_operator.py

```diff
@@ -1,482 +1,80 @@
-        "namespace": "namespace",
-        "port": 1234,
-        "secret-key": "minio-super-secret-key",
-        "secure": True,
-        "service": "service",
-    }
-    os_data = {"_supported_versions": "- v1", "data": yaml.dump(os_data_dict)}
-    os_rel_id = harness.add_relation("object-storage", "storage-provider")
-    harness.add_relation_unit(os_rel_id, "storage-provider/0")
-    harness.update_relation_data(os_rel_id, "storage-provider", os_data)
-
-    # ingress
-    ingress_relation_name = "ingress"
-    relation_version_data = {"_supported_versions": "- v1"}
-    ingress_rel_id = harness.add_relation(
-        ingress_relation_name, f"{ingress_relation_name}-subscriber"
-    )
-    harness.add_relation_unit(ingress_rel_id, f"{ingress_relation_name}-subscriber/0")
-    harness.update_relation_data(
-        ingress_rel_id, f"{ingress_relation_name}-subscriber", relation_version_data
-    )
-
-    # Mock away _validate_default_s3_bucket to avoid using boto3/creating clients
-    mocked_validate_default_s3_bucket = mocker.patch("charm.Operator._validate_default_s3_bucket")
-    bucket_name = harness._backend.config_get()["default_artifact_root"]
-    mocked_validate_default_s3_bucket.return_value = bucket_name
-
-    # pod defaults relations setup
-    pod_defaults_rel_name = "pod-defaults"
-    pod_defaults_rel_id = harness.add_relation(
-        "pod-defaults", f"{pod_defaults_rel_name}-subscriber"
-    )
-    harness.add_relation_unit(pod_defaults_rel_id, f"{pod_defaults_rel_name}-subscriber/0")
-
-    harness.begin_with_initial_hooks()
-
-    pod_spec = harness.get_pod_spec()
-    yaml.safe_dump(pod_spec)
-    assert harness.charm.model.unit.status == ActiveStatus()
-
-    assert pod_spec[0]["kubernetesResources"]["services"] == []
-
-
-def test_storage_endpoint_generation(harness, sample_object_storage):
-    """Test S3 storage URL for correct generation in S3 service and in charm code."""
-    harness.begin()
-    s3_wrapper = S3BucketWrapper(
-        access_key=sample_object_storage["access-key"],
-        secret_access_key=sample_object_storage["secret-key"],
-        s3_service=f"{sample_object_storage['service']}.{sample_object_storage['namespace']}",
-        s3_port=sample_object_storage["port"],
+    @patch(
+        "charm.KubernetesServicePatch",
+        lambda x, y, service_name, service_type, refresh_event: None,
     )
+    def tests_on_pebble_ready_success(self, harness: Harness):
+        harness.begin()
+        harness.charm._on_event = MagicMock()
+        harness.charm._on_pebble_ready(None)
+        harness.charm._on_event.assert_called()
 
-    assert s3_wrapper.s3_url == harness.charm._gen_obj_storage_endpoint_url(
-        obj_storage=sample_object_storage
+    @patch(
+        "charm.KubernetesServicePatch",
+        lambda x, y, service_name, service_type, refresh_event: None,
     )
+    def test_update_layer_success(
+        self,
+        harness: Harness,
+    ):
```

Comment:
Is there a better name we can give this?  It tests more than just "update_layer_success"